python/ray/autoscaler/_private/commands.py

```diff
@@ -757,6 +749,59 @@ def get_or_create_head_node(config: Dict[str, Any],
         cli_logger.print("  {}", remote_shell_str.strip())
 
 
+def _should_create_new_head(head_node: Optional[str], launch_hash: str,
+                            head_node_type: str,
```

Comment:
The only new logic here is the node type name check. Just extracted the if condition into its own function + added the extra check.